ppoprf/src/ppoprf.rs

```diff
@@ -100,6 +257,34 @@ impl ProofDLEQ {
     RistrettoScalar::from_bytes_mod_order_wide(&out)
   }
 
+  // https://datatracker.ietf.org/doc/html/draft-irtf-cfrg-hash-to-curve-16#name-hash_to_field-implementatio
+  // The hash_to_field function is also suitable for securely hashing
+  // to scalars. For example, when hashing to the scalar field for an
+  // elliptic curve (sub)group with prime order r, it suffices to
+  // instantiate hash_to_field with target field GF(r).
+  fn hash_to_scalar(input: &[u8], label: &str) -> RistrettoScalar {
+    let mut uniform_bytes = [0u8; DIGEST_LEN];
+    // strobe_hash() -> expand_msg_xmd
+    // strobe_hash uses Keccak internally
+    strobe_hash(input, label, &mut uniform_bytes);
+
+    // convert hash output to scalar
+    ProofDLEQ::os2ip(&uniform_bytes)
+  }
+
+  // Convert a byte sequence into a scalar
+  fn os2ip(input: &[u8; 64]) -> RistrettoScalar {
+    RistrettoScalar::from_bytes_mod_order_wide(input)
+  }
+
+  // I2OSP2(x): Converts a non-negative integer x into a byte
+  // array of length 2 as described in [RFC8017]. Note that
+  // this function returns a byte array in big-endian byte order.
+  fn i2osp2(x: usize) -> [u8; 2] {
```

Comment:
This should probably be generalised to a any len, and not only two, as in https://github.com/cfrg/draft-irtf-cfrg-hash-to-curve/blob/6d40f9890a5eb63ebfa0da9e49678e140e11f755/poc/hash_to_field.py#L25

Reply:
Heh. Originally it was, but the algorithm only ever used the 2-byte variant, so I suggested the specialization to simplify error handling.